lib/state_machine/types.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "message.hpp"
+#include "state.hpp"
+
+#include <cstdint>
+
+#include "utils/hash.hpp"
+
+namespace hyped::state_machine {
+struct SourceAndMessage {
+  State source;
+  Message message;
+
+  bool operator==(const SourceAndMessage &key) const
+  {
+    return key.source == source && key.message == message;
+  }
+};
+}  // namespace hyped::state_machine
+
+namespace std {
+template<>
+struct hash<hyped::state_machine::SourceAndMessage> {
```

Comment:
redundant `hyped`

Reply:
And elsewhere in the struct.

Reply:
Then we'll have to change the job to check for redundant hyped since this currently causes it to fail.

Reply:
@kshxtij :joy: